04-Lab/FunctionalSpatialImageProcessing.ipynb

```diff
@@ -0,0 +1,517 @@
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# We need to invert our warp from T1_2_MNI to get the MNI_2_T1 warp\n",
+    "# Generic usage: invwarp -w A2B_warp -o B2A_warp -r A\n",
+    "invwarp -w T1_2_MNI_warp -o MNI_2_T1_warp -r my_T1_brain.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# We can invert our func2T1_6dof.mat with convert_xfm\n",
+    "# Try this below, name the inverted matrix T12func.mat\n",
+    "convert_xfm -omat T12func_6dof.mat -inverse func2T1_6dof.mat"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Extracting an atlas-based region of interest in MNI space\n",
+    "# Copy the Harvard-Oxford sub-cortical atlas from fsl's library to our working directory\n",
+    "cp /usr/share/fsl/data/atlases/HarvardOxford/HarvardOxford-sub-maxprob-thr50-2mm.nii.gz HarvardOxford-sub-maxprob-thr50-2mm.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "# change the image type to Mask/Label in fslview (remember where to look?)\n",
+    "# also change the colorscale from grayscale to render or random-rainbow\n",
+    "fslview HarvardOxford-sub-maxprob-thr50-2mm.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Example for extracting the left putamen (intensity 6)\n",
+    "# review: what does -bin do?\n",
+    "fslmaths HarvardOxford-sub-maxprob-thr50-2mm.nii.gz -thr 6 -uthr 6 -bin Lputamen.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Did it work?\n",
+    "fslview MNI152_T1_2mm_brain.nii.gz Lputamen.nii.gz"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# Finally we can use apply warp to apply both our nonlinear and affine warps at once to our functional\n",
+    "# This is typically done when we want to put a regional mask in standard space into functional space \n",
+    "# e.g., for our atlas-based ROI of the caudate\n",
+    "# interp=nn means the interpolation method we used is nearest neighbor.\n",
+    "# nearest neighbor: this means a voxel takes the value of it's closest neighbor\n",
+    "applywarp -i Lputamen -r example_func_brain -o Lputamen_mask_func -w MNI_2_T1_warp --postmat=T12func_6dof.mat --interp=nn"
```

Comment:
I used nearest neighbor here because it's cleaner (fewer intermediate files) and removes one permutation from the already complex landscape of mri analysis (do I threshold by 0.5, 0.6, 0.4 etc.), but if we want greater control over the threshold we can keep the original method and threshold from that.